### PR TITLE
HIP-584: Disable allowance ERC precompile

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:

This PR disables the `allowance(address token, address owner, address spender) external returns (int responseCode)` precompile by throwing `UnsupportedOperationException` if it is accessed. The reason is, there is some incorrect data in (fixed in 37 services release), connected to exporting child records for approve operations related to fungible tokens.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java [hunk 1]
```diff
@@ -72,6 +72,9 @@ public class MirrorNodeEvmProperties implements EvmProperties {
     @NotNull
     private HederaNetwork network = HederaNetwork.TESTNET;
 
+    @Getter
+    private boolean isAllowanceEnabled = false;
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:76]: Lombok should generate the `is` prefix for the getter.
```suggestion
    private boolean allowanceEnabled = false;
```

Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:76]: Sort and add to docs/configuration.md

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/token/TokenAccessorImpl.java [hunk 2]
```diff
@@ -218,12 +218,16 @@ public long balanceOf(Address account, Address token) {
 
     @Override
     public long staticAllowanceOf(final Address owner, final Address spender, final Address token) {
-        final var tokenAllowanceId = new AbstractTokenAllowance.Id();
-        tokenAllowanceId.setOwner(entityIdFromAccountAddress(owner));
-        tokenAllowanceId.setSpender(entityIdFromAccountAddress(spender));
-        tokenAllowanceId.setTokenId(entityIdNumFromEvmAddress(token));
+        if(properties.isAllowanceEnabled()) {
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/token/TokenAccessorImpl.java:221]: Prefer to fail fast and keep the other code as unchanged.
```suggestion
        if (!properties.isAllowanceEnabled()) {
            throw new UnsupportedOperationException("allowance(address owner, address spender) is not supported.");
        }
```

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceERCTokenTest.java [hunk 3]
```diff
@@ -86,11 +86,20 @@ void metadataOf() {
...
+
+        assertThatThrownBy(() -> contractCallService.processCall(serviceParameters)).
+                isInstanceOf(UnsupportedOperationException.class).hasMessage("allowance(address owner, address spender) is not supported.");
+    }
+
     @RequiredArgsConstructor
     public enum ContractFunctions {
         GET_APPROVED_EMPTY_SPENDER("getApproved",new Object[] {NFT_ADDRESS, 2L}, new Address[] {Address.ZERO}),
         IS_APPROVE_FOR_ALL        ("isApprovedForAll", new Address[] {NFT_ADDRESS, SENDER_ADDRESS, RECEIVER_ADDRESS}, new Boolean[] {true}),
-        ALLOWANCE_OF              ("allowance", new Address[] {FUNGIBLE_TOKEN_ADDRESS, SENDER_ADDRESS, RECEIVER_ADDRESS}, new Long[] {13L}),
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceERCTokenTest.java:93]: Think we should keep this test, and in the test that uses it set `properties.setAllowanceEnabled(true)`.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]

**Pull request merged**